dev-docs/RFCs/v7.x/binary-data-rfc.md

```diff
@@ -21,15 +21,103 @@ This article focused on the following use cases:
 * **Using typed arrays as layer input data** - Using The input data is in binary form, perhaps it is delivered this way from the back-end, and it would be preferable to not have to unpack it.
 * **Using typed arrays or GPU buffers directly as layer attributes** - The input data is already formatted in the memory format expected by the GPU and deck.gl's shaders.
 
-## Using Flattend Data (Typed Arrays as layer input data
+## Using Flattend Data (Typed Arrays as layer input data)
 
-In some cases the application may receive all or parts of a layer's geometry in the form of binary typed arrays instead of Javascript arrays. Typed arrays can sometimes be transported more efficiently in binary form than classic arrays that tend to be JSON stringified and then parsed.
+In some cases the application may receive all or parts of a layer's geometry in the form of binary typed arrays instead of Javascript arrays. When communicating with a server or web workers, typed arrays can be transported more efficiently in binary form than classic arrays that tend to be JSON stringified and then parsed. For example, a ScatterplotLayer may receive input data like this:
 
-One of the biggest differences that makes it hard to use binary arrays directly is that they are flat, whereas JavaScript arrays tend to have a sub-array for each vertex or color.
+```js
+// lon1, lat1, radius1, red1, green1, blue1, lon2, lat2, ...
+const binaryData = new Float32Array([-122.4, 37.78, 1000, 255, 200, 0, -122.41, 37.775, 500, 200, 0, 0, -122.39, 37.8, 500, 0, 40, 200]);
+```
+
+Upon receiving the typed arrays, we can of course re-construct a classic JavaScript array:
 
-TBA...
+```js
+const data = [];
+for (let i = 0; i < binaryData.length; i += 6) {
+  data.push({
+    position: binaryData.subarray(i, i + 2),
+    radius: binaryData[i + 2],
+    color: binaryData.subarray(i + 3, i + 6)
+  });
+}
+
+new ScatterplotLayer({
+  data,
+  getPosition: d => d.position,
+  getRadius: d => d.radius,
+  getColor: d => d.color
+});
+```
 
-deck.gl layers carefully formats data in memory so that it can be accessed efficiently by the GPU...
+However, this array will take valuable CPU time to create and significantly more memory to store than its binary form. In performance-sensitive applications that constantly push a large volumn of data (e.g. animations), this method will not be efficient enough.
+
+Alternatively, one may supply a non-iterable object (not Array or TypedArray) to the `data` object. In this case, it must contain a `length` field that specifies the total number of objects. Since `data` is not iterable, each accessor will not receive a valid `object` argument, and therefore responsible of interpreting the input data's buffer layout:
```

Comment:
Add header and wording changes?

### Proposal

"Alternatively, one may supply a non-iterable object..." 
=> 
"This RFC proposes that if the source data is not in the form of a single array with each index corresponding to a visual element, one may instead supply an object that describes how the data is broken down into separate arrays..."

Reply:
At the top of this doc:

> This RFC is written as an article for the deck.gl Developer's Guide, rather than in the standard format. The intention is to copy this text to the developer's guide once it matures.